geom.go

```diff
@@ -1,14 +1,18 @@
 // Package geom describes geometry interfaces.
 package geom
 
-type Geometry interface{}
+// Geometry is an object with a spatial reference.
+type Geometry interface {
+	Points() [][2]float64
```

Comment:
Well, I did build this capability as part of Beachfront but it is not currently part of the product. I don't recall what I needed it for off of the top of my head. 

I submit that we should have some level of encapsulation in these structs. In this case it means having BBox() and/or Points() as part of Geometry.

Reply:
Alex and I had this conversion. I don't think we should have Point in the Geometery Interface, and if we do want to have it there, it should be named something else. The way you have it currently, make it hard to describe the intent of the function when working with LineStrings, Points or Polygons or Geometries; since they all have the same Func signature. Where these are semantically different things and should be treated differently. Making it difficult for a ClipLineString function to assume certain things.

Reply:
A ClipLineString should only operate on one-dimensional (or maybe two dimensional) geometries and the only way to find out a geometry's dimensionality is to ask. This function is currently missing from the interface but I believe it needs to be there.

Reply:
I don't see how you don't have that with the interface. The fact that SubPoints() returns [][2]float64, tell you that you have a line made up of two-dimensional points.  And the fact that it's the SubPoints() method you know these points are meant to be connected.  Whereas if it's just Point() like for MultiPointer then you know that the points are disconnected.